src/suggestion.rs

```diff
@@ -28,6 +31,49 @@ pub enum Detector {
     Dummy = 0b1000,
 }
 
+pub fn get_terminal_size() -> usize {
+    use super::*;
+    const DEFAULT_TERMINAL_SIZE: usize = 80;
+    match crossterm::terminal::size() {
+        Ok((terminal_size, _)) => terminal_size as usize,
+        Err(_) => {
+            warn!(
+                "Unable to get terminal size. Use default: {}",
+                DEFAULT_TERMINAL_SIZE
+            );
+            DEFAULT_TERMINAL_SIZE
+        }
+    }
+}
+
+pub fn get_current_statement<'a>(arr: &'a Vec<&'_ str>, range: Range) -> (&'a str, usize) {
+    let mut stripped_line: &str = "";
+    let mut initial_sentence: usize = 0;
+    let mut line_pos: usize = 0;
+    for (pos, sentence) in arr.iter().enumerate() {
+        initial_sentence += sentence.chars().count();
+        line_pos = pos;
+        stripped_line = sentence;
+        if range.end < initial_sentence {
+            break;
+        }
+    }
+    (stripped_line, line_pos)
+}
+
+pub fn char_sub_window(s: &str, range: Range) -> &str {
+    // @todo can be done in a single iterator, use `fold()`
```

Comment:
This seems to be equiv to the `struct` associated `fn` with the same name?

Reply:
removed `fn` explained above why.